gammapy/analysis/core.py

```diff
@@ -293,262 +285,110 @@ def _extract_irf_kernels(self, dataset):
+        e_reco = self._make_energy_axis(datasets_settings.geom.axes.energy).edges
         maker_config["e_reco"] = e_reco
-
         # TODO: remove hard-coded e_true and make it configurable
         maker_config["e_true"] = np.logspace(-2, 2.5, 109) * u.TeV
         maker_config["region"] = on_region
 
         dataset_maker = SpectrumDatasetMaker(**maker_config)
         bkg_maker_config = {}
-        background = self.settings["datasets"]["background"]
-
-        if "exclusion_mask" in background:
-            map_hdu = {}
-            filename = background["exclusion_mask"]["filename"]
-            if "hdu" in background["exclusion_mask"]:
-                map_hdu = {"hdu": background["exclusion_mask"]["hdu"]}
-            exclusion_region = Map.read(filename, **map_hdu)
+        if datasets_settings.background.exclusion:
+            exclusion_region = Map.read(datasets_settings.background.exclusion)
             bkg_maker_config["exclusion_mask"] = exclusion_region
-        if background["background_estimator"] == "reflected":
-            reflected_bkg_maker = ReflectedRegionsBackgroundMaker(**bkg_maker_config)
-        else:
-            # TODO: raise error?
-            log.info("Background estimation only for reflected regions method.")
-
+        bkg_maker = ReflectedRegionsBackgroundMaker(**bkg_maker_config)
         safe_mask_maker = SafeMaskMaker(methods=["aeff-default", "aeff-max"])
 
         datasets = []
         for obs in self.observations:
             log.info(f"Processing observation {obs.obs_id}")
             selection = ["counts", "aeff", "edisp"]
             dataset = dataset_maker.run(obs, selection=selection)
-            dataset = reflected_bkg_maker.run(dataset, obs)
+            dataset = bkg_maker.run(dataset, obs)
             if dataset.counts_off is None:
-                log.info(
-                    f"No OFF region found for observation {obs.obs_id}. Discarding."
-                )
+                log.info(f"No OFF region found for observation {obs.obs_id}. Discarding.")
                 continue
             dataset = safe_mask_maker.run(dataset, obs)
             log.debug(dataset)
             datasets.append(dataset)
 
         self.datasets = Datasets(datasets)
-
-        if self.settings["datasets"]["stack-datasets"]:
+        if self.config.datasets.stack:
             stacked = self.datasets.stack_reduce()
             stacked.name = "stacked"
             self.datasets = Datasets([stacked])
 
+    @staticmethod
+    def _make_energy_axis(axis):
+        """Helper method to build an energy axis."""
+        axis_params = dict(name="energy",
+                           lo_bnd=axis.min.value,
+                           hi_bnd=axis.max.value,
+                           nbin=axis.nbins,
+                           unit=axis.min.unit,
+                           interp="log",
+                           node_type="edges")
+        energy_axis = MapAxis.from_bounds(**axis_params)
```

Comment:
Suggest to rewrite like this, since the local `axis_params` and `energy_axis` variables provide no extra value:
```
return MapAxis.from_bounds(
    name="energy",
    ....
)
```

Reply:
Done